app/controllers/AlertController.java

```diff
@@ -61,6 +72,89 @@ public AlertController(final Configuration configuration, final AlertRepository
     }
 
     /**
+     * Adds an alert in the alert repository.
+     *
+     * @return Created if the alert was created successfully, an error code otherwise.
+     */
+    public Result add() {
+        Alert alert;
+        try {
+            final models.view.Alert viewAlert = buildViewAlert(request().body());
+            alert = convertToInternalAlert(viewAlert);
+        } catch (final IOException | ConstraintsViolatedException | IllegalArgumentException e) {
+            LOGGER.error()
+                    .setMessage("Failed to build an alert.")
+                    .setThrowable(e)
+                    .log();
+            return badRequest("Invalid request body.");
+        }
+
+        if (_alertRepository.get(alert.getId()).isPresent()) {
+            LOGGER.error()
+                    .setMessage(String.format("An alert with id %s already exists", alert.getId().toString()))
+                    .log();
```

Comment:
Granted this statement will be removed with the above change, but please consider that this should not be an error. The client sent bad data for the operation intended. At most a warning, I could even believe it's an info. Second, you should never format messages into the structured log builder. The point of the builder is to accept supporting data and context in such a way that it is separate and parseable from the message.